roles/signals_test.py

```diff
@@ -0,0 +1,85 @@
+from factory.django import mute_signals
+from rolepermissions.verifications import (
+    has_role,
+    has_permission,
+    has_object_permission,
+)
+
+from courses.factories import ProgramFactory
+from roles.models import Role
+from profiles.factories import UserFactory
+
+
+# pylint: disable=no-self-use
+
+class SignalsTest(TestCase):
+    """
+    Tests for signals triggered by the role assignment
+    """
+
+    def setUp(self):
+        super(SignalsTest, self).setUp()
+        self.user = UserFactory.create()
+        self.program = ProgramFactory.create()
+
+    def test_assign_role(self):
+        """
+        Assigning the role with the model triggers a signal
+        to assign the correspondent role to the user.
+        """
+        assert has_role(self.user, 'staff') is False
+        assert has_permission(self.user, 'can_advance_search') is False
+        assert has_object_permission('can_advance_search', self.user, self.program) is False
+        Role.objects.create(
+            program=self.program,
+            user=self.user,
+            role='staff',
+        )
+        assert has_role(self.user, 'staff') is True
+        assert has_permission(self.user, 'can_advance_search') is True
+        assert has_object_permission('can_advance_search', self.user, self.program) is True
+
+    def test_presave_removes_current_role(self):
+        """
+        Updating the role in the model triggers a pre_save
+        signal that removes the previous role from the user.
+        """
+        mm_role = Role.objects.create(
+            program=self.program,
+            user=self.user,
+            role='staff',
+        )
+        assert has_role(self.user, 'staff') is True
+        assert has_permission(self.user, 'can_advance_search') is True
+        assert has_object_permission('can_advance_search', self.user, self.program) is True
+
+        with mute_signals(post_save):
+            mm_role.role = 'instructor'
+            mm_role.save()
+        assert has_role(self.user, 'staff') is False
+        assert has_permission(self.user, 'can_advance_search') is False
+        assert has_object_permission('can_advance_search', self.user, self.program) is False
+
+    def test_postdelete_removes_role(self):
+        """
+        If role assignment is removed, a signal takes
+        care of removing the role from the user.
+        """
+        mm_role = Role.objects.create(
+            program=self.program,
+            user=self.user,
+            role='staff',
+        )
+        assert has_role(self.user, 'staff') is True
+        assert has_permission(self.user, 'can_advance_search') is True
+        assert has_object_permission('can_advance_search', self.user, self.program) is True
+
+        mm_role.delete()
+        assert has_role(self.user, 'staff') is False
+        assert has_permission(self.user, 'can_advance_search') is False
+        assert has_object_permission('can_advance_search', self.user, self.program) is False
```

Comment:
i think it's probably worth it to include a test case that removes a user's role for one program, but they still have that role for a different program